Source/CompSpinalEngineTrail.cs

```diff
@@ -0,0 +1,179 @@
+﻿using RimWorld;
+using SaveOurShip2;
+using System.Text;
+using Verse;
+
+namespace TheCafFiend
+{
+    [StaticConstructorOnStartup]
+    public class CompSpinalEngineTrail : CompEngineTrail
+    {
+        public override string CompInspectStringExtra()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            
+            if (fullyFormed)
+            {
+                stringBuilder.Append($"Fuel burn rate per second: {cachedFuelUse} of {cachedFuelAllowed} supported");
+                stringBuilder.AppendInNewLine($"Thrust: {cachedThrust * 500}");
+            }
+            else
+            {
+                stringBuilder.Append(cachedError.Colorize(UnityEngine.Color.red));
+            }
+            return stringBuilder.ToString();
+        }
+        public virtual CompProperties_SpinalEngineTrail Properties //Wait is this dirty I just want to be clear it's not hiding hrm
+        {
+            get { return props as CompProperties_SpinalEngineTrail; }
+        }
+        public class ReturnValue
+        {
+            public float thrustAmp;
+            public float fuelAmp;
+            public float fuelAllowAmp;
+            public float powerUseAmp;
+            public int supportWeight;
+            public string playerError;
+            public bool fullyFormed;
+        }
+        private int cachedThrust = 0;
+        private int cachedMass = 0; // add/remove on the *engine* does this itself, this is only amps+caps
+        public bool fullyFormed = false;
+        private int cachedFuelUse = 0;
+        private int cachedPowerUse = 0;
+        private int cachedFuelAllowed = 0; // Error below probably never surfaces to a player but I want a fallback
+        private string cachedError = "Spinal engine isn't fully formed! Check for engine accelerators, and all three components of the fuel infrastructure!";
+        public override int Thrust
+        {
+            get
+            {
+                if (parent.TryGetComp<CompSpinalEngineTrail>() == null) //should never? happen but reflex
+                {
+                    Log.Message("SOS2 spinal engines: somehow hit thrust from a non-compspinalenginetrail?");
+                    return base.Thrust;
+                }
+                if (fullyFormed)
+                {
+                    //Log.Message($"sos2 spinal engines thrust called about a spinal engine, returning cached thrust {thrustCached}");
+                    return cachedThrust;
+                }
+                else
+                {
+                    //Log.Message($"sos2 spinal engines thrust called about a spinal engine, not fully formed");
+                    calc();
+                    return cachedThrust;
+                }
+            }
+        }
+        public string CurrentError // Feels a little silly to have a setter just for this, maybe cachedError should just be public? 
```

Comment:
That's a good question! Currently, this setter is only used in one place anyhow, thus the "Should I just make the error public?" I'm not very married to any approach here; A simple function, a plain public cachedError, I guess the setter is... Essentially? the same as a dedicated function? Doesn't currently need any handling, it would be more of a questuion of what feels more future-proof vs easy to read

Reply:
Yeah, you'd use an actual function there.  Traditionally I'd say error handling is better done as a function, but there's no reason you can't do it this way.  It's a bit unusual in my mind to see just the setter for the property that sets the variable but it's not like the function.

Reply:
Yeah upon reflection, the current implementation can just be a function that sets the variable, but if it's a real function it's easier to extend later if it needs fancy handling. I mean I could extend a setter later I... guess, but that would feel very illegal and unanticipated